### ISSUE TITLE
Vue Apps mounting but never unmounting

### ISSUE THREAD

**Issue by seancrater** [retejs/vue-render-plugin#73]
Hey there!

We have an app where we're using custom Vue nodes and rendering by passing Rete a ref to a `div` element but we're running into a little bit of a hiccup that seems to be caused by the way the Vue nodes, connections, and sockets are mounted. They successfully mount and Rete renders correctly but it seems these nodes, connections, and sockets are never cleaned up by Rete no matter which Rete lifecycle events I fire. Currently, in `onBeforeUnmount`, I'm firing a `destroy` method that I return in our Rete instance that fires two Rete cleanup methods:
```
    destroy: async () => {
      await editor.clear()
      area.destroy()
    },
```

Unfortunately, neither of these seem to remove the Vue apps that are mounted during the initial render of Rete.
![image](https://github.com/retejs/vue-plugin/assets/6605678/7f929d8a-9ffa-4247-ab62-1f31cdf82022)

TL;DR The behavior I'd expect is that these Vue apps are unmounted when Rete is unmounted but that doesn't seem to be the case.

## Additional Info
We're using the standard Rete config for custom Vue components as far as I can tell:
```
  render.addPreset(
    Presets.classic.setup({
      customize: {
        node() {
          return IssueVisualizationNode
        },
        connection() {
          return IssueVisualizationConnection
        },
        socket() {
          return IssueVisualizationSocket
        },
      },
    }),
  )
```

One thing that could be impacting this is how we create our nodes and links so to share, we are creating them in this code block dynamically using an API response:
```
  const nodesById: { [key: string]: Node } = {}

  // We build out each individual node and set a reference to each in nodesById
  // for when we build out the connections in the following loop
  const nodeRequests = props.data.nodes.map((node: { id: number }) => {
    const nodeId = `${node.id}`
    const newNode = createNode(nodeId, socket)
    newNode.customAttributes = { ...node }
    nodesById[nodeId] = newNode

    return editor.addNode(newNode)
  })

  // Await all node requests to resolve before proceeding to building connections
  await Promise.all(nodeRequests)

  // Build out connections - the lines between nodes in the visualization
  const connectionRequests = props.data.links.map((link: { from: number; to: number }) => {
    const { from, to } = link
    const fromNode = nodesById[from]
    const toNode = nodesById[to]

    const connection = new ClassicPreset.Connection(
      fromNode as ClassicPreset.Node,
      'port',
      toNode as ClassicPreset.Node,
      'port',
    ) as Connection

    // Bind all connection (aka link) data for consumption in the connection component
    connection.linkAttributes = link

    return editor.addConnection(connection)
  })

  await Promise.all(connectionRequests)
```

**Comment by Ni55aN** [comment]
The first thing you need to do is make sure that [this line](https://github.com/retejs/vue-plugin/blob/main/src/index.ts#L44C14-L44C21) is called (in Devtools' Source tab with enabled Source maps) for every node, control, and socket

There are several reasons why it is not called:
1. the node is not deleted correctly (`editor.clear` -> `editor.removeNode` -> `noderemoved` event trigger in the core -> `unmount` event trigger in the area plugin -> `unmount` method in the vue plugin)
2. using outdated version of the plugin where <Ref> component doesn't unmount controls and sockets

btw I cannot reproduce it on the basic application created using Rete Kit (customization template)
```ts
  setTimeout(async () => {
      await editor.clear()
      area.destroy()
  }, 5000)
```
![vue](https://github.com/retejs/vue-plugin/assets/8259641/c7f2ade6-f547-4cd2-b17c-ec04aa578892)

**Comment by seancrater** [comment]
Thank you for the response @Ni55aN! Validated that the method isn't being called so I'll do some digging on my end of what I could be doing wrong. 😄

**Comment by seancrater** [comment]
Turned out to be a memory leak caused by un-mounting/re-mounting the component that was instantiating Rete via a `v-if` conditional block, it wasn't firing the `onUnmount` lifecycle hook
